tempodb/backend/raw.go

```diff
@@ -222,27 +243,57 @@ func (r *reader) BlockMeta(ctx context.Context, blockID uuid.UUID, tenantID stri
 
 // TenantIndex implements backend.Reader
 func (r *reader) TenantIndex(ctx context.Context, tenantID string) (*TenantIndex, error) {
-	reader, size, err := r.r.Read(ctx, TenantIndexName, KeyPath([]string{tenantID}), nil)
+	ctx, span := tracer.Start(ctx, "reader.TenantIndex")
+	defer span.End()
+
+	outPb, err := r.tenantIndexProto(ctx, tenantID)
+	if err == nil {
```

Comment:
Wonder if we want to be more strict in the error checks here, by only falling back to json if the proto error is `backend.ErrDoesNotExist` ?   I don't think it makes sense to try the json if there was a proto index but it was corrupted. In that case it would be better to surface the error and fail.

Reply:
I've pushed a commit for this.